lighthouse/classes/events/beckman/source_all_negatives.py

```diff
@@ -0,0 +1,52 @@
+import logging
+from typing import Any, Dict
+
+from lighthouse.classes.event_properties.definitions import (
+    PlateBarcode,
+    SourcePlateUUID,
+    UserID,
+    RobotSerialNumber,
+)
+from lighthouse.classes.event_properties.definitions.beckman import RobotUUID, PositiveSamplesFromSource
+from lighthouse.classes.events import PlateEvent
+from lighthouse.classes.services.labwhere import LabwhereServiceMixin
+
+logger = logging.getLogger(__name__)
+
+
+class SourceAllNegatives(PlateEvent, LabwhereServiceMixin):
+    def __init__(self, event_type: str) -> None:
+        super().__init__(event_type=event_type, plate_type=PlateEvent.PlateTypeEnum.SOURCE)
+        self.properties: Dict[str, Any] = {}
+
+    def initialize_event(self, params: Dict[str, str]) -> None:
+        super().initialize_event(params=params)
+
+        self._event_type = params["event_type"]
+        self.properties["plate_barcode"] = PlateBarcode(params)
+
+        for property_name in ["plate_barcode"]:
+            self.properties[property_name].is_valid()
+
+        self.properties["source_plate_uuid"] = SourcePlateUUID(self.properties["plate_barcode"])
+        self.properties["positive_samples_from_source"] = PositiveSamplesFromSource(
+            self.properties["source_plate_uuid"]
+        )
+        self.properties["user_id"] = UserID(params)
+        self.properties["robot_serial_number"] = RobotSerialNumber(params)
+        self.properties["robot_uuid"] = RobotUUID(self.properties["robot_serial_number"])
```

Comment:
Maybe we can move this code to line 26 (before the checks that happen in: 
```python
        for property_name in ["plate_barcode"]:
            self.properties[property_name].is_valid()
```
because we can validate if the input is right before accessing of other services.